### PR TITLE
added the codeql.yml

### PR DESCRIPTION
**Pull request by KrishnaSindhur**
#1291

Added the codeql checks

### CODE REVIEW

**Comment by KrishnaSindhur** [comment]
@jpkrohling  can u review this, its my first PR in this repo

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+          go-version: 1.14
+        id: go
+
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v2
+
+      - name: Test
+        run: go test -v ./...
+
+security:
```
Comment (jpkrohling) [inline .github/workflows/codeql.yml:26]: Looks like this isn't properly formatted and the above action doesn't seem appropriate for this repository. Note how this check wasn't added as part of the checks for this PR.

Comment (KrishnaSindhur) [reply .github/workflows/codeql.yml]: i will check that

**Comment by jpkrohling** [comment]
Could you please take a look at the DCO check?

**Comment by KrishnaSindhur** [comment]
> Could you please take a look at the DCO check?

sure done

File: .github/workflows/codeql.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+  schedule:
+    - cron: '31 6 * * 1'
```
Comment (jpkrohling) [inline .github/workflows/codeql.yml:9]: Why do we want to run this on a periodic basis? Should be sufficient to run on PRs and on merge events for master, no?

Comment (KrishnaSindhur) [reply .github/workflows/codeql.yml:9]: ya i had that doubt, should we run this once in a day, week or only during PRs and on merge, so should i remove the cron?

**Review by jpkrohling — APPROVED** [review]

**Pull request merged**